source/features/link-to-gihub-io.tsx

```diff
@@ -0,0 +1,88 @@
+import React from "dom-chef";
+import elementReady from "element-ready";
+import select from "select-dom";
+import * as pageDetect from "github-url-detection";
+import LinkExternalIcon from "octicon/link-external.svg";
+
+import features from ".";
+
+function getCurrentUsername(): string {
+	return location.pathname.replace(/(^\/)/g, "").split("/")[0];
+}
+
+function getGithubIoLink(username: string): string {
+	return `${username}.github.io`;
+}
+
+async function repoTabInit(): Promise<void> {
+	const username = getCurrentUsername();
+	const githubIoLink = getGithubIoLink(username);
+	const repo = select(`a.rgh-discussion-links[href$="${githubIoLink}"]`);
+	repo?.parentElement?.appendChild(
+		<a
+			href={`https://${githubIoLink}`}
+			target="black"
+			rel="noopener noreferrer"
+		>
+			<LinkExternalIcon />
+		</a>
+	);
+	return;
+}
+
+async function githubIoRepoInit(): Promise<void> {
+	const repositoryName = pageDetect.utils.getRepoURL().split("/")[1];
+	const githubIoLink = getGithubIoLink(getCurrentUsername());
+	const link = (await elementReady(".BorderGrid-cell .octicon-link"))!;
```

Comment:
We use `elementReady` to look for an element **before** `dom-loaded` and edit it before the first paint, if possible.

Since we expect the link to **not** be there, `elementReady` here will just wait until `dom-loaded` before stopping. 

Either use a plain `select(…)` or use `elementReady` the link container **and then** look for the link, e.g.:

```js
if (select.exists('.octicon-link', await elementReady('.BorderGrid-cell'))) {
  return false;
}

// else, add the link 
```